sklearn/mixture/_base.py

```diff
@@ -120,15 +120,15 @@ def _initialize_parameters(self, X, random_state):
             resp[np.arange(n_samples), label] = 1
         elif self.init_params == "random":
             resp = random_state.uniform(size=(n_samples, self.n_components))
-            resp /= resp.sum(axis=1)[:, np.newaxis]
+            resp /= resp.sum(axis=1)[:, np.newaxis].astype(X.dtype)
```

Comment:
I am not sure it matters that much, but it seems like `resp` is always going to be float64 (apparently `.uniform` does not let you specify a `dtype`). In particular the `.astype(X.dtype)` is not going to do anything?

Reply:
Indeed! TIL. If `resp.dtype` is not `float32` then dividing by a `float32` won't change the dtype of it :-/

I guess we'd have to add a `resp = resp.astype(X.dtype)` and maybe can remove the `astype` from `resp /= resp.sum(axis=1)[:, np.newaxis].astype(X.dtype)`?

Reply:
Yes this was a valid point. Thanks for catching it @lesteve 
I updated this and also added global_dtype in a test that uses this particular initialization.